### DISCUSSION TITLE
Error: cannot read properties of undefinded/0 because array is empty

### DISCUSSION THREAD

**Discussion by HenrikeStahlhut** [neuefische/questions#619] [Web] [answered]
Hi everyone, 

**introduction:** this is the continuation of my last problem, [here's the discussion](https://github.com/orgs/neuefische/discussions/617). 
I fixed my previous problem with `$lookup`, but now i ran into another bug. 

In theory, the logic works. I now have an additional array called `rooms` in every plant object, which contains the `title`, `id` and `roomtype` of the respective room the plant is in:
 
![Screenshot 2023-05-25 at 10 58 29](https://github.com/neuefische/questions/assets/126799897/0aab21e1-ac44-4c58-b5c0-7547bf460dd9)

Now i can display each plants room with the following code: 
```
return (
    <>
      <StyledList>
        {plants.map((plant) => (
          <StyledCard key={plant._id}>
            <StyledListItem key={plant._id}>
              🪴 {plant.title} - Room:{plants.rooms[0].title}
            </StyledListItem>
          </StyledCard>
        ))}
      </StyledList>
    </>
  );
```

**The problem:** If i delete a room, the array `rooms` is now empty, therefore `Room:{plants.rooms[0].title}` throws an error, because the array is 0/undefined. 

![Screenshot 2023-05-25 at 11 03 15](https://github.com/neuefische/questions/assets/126799897/c8b121cc-751c-415c-8f8f-ae84c30bdea8)
![Screenshot 2023-05-25 at 11 03 29](https://github.com/neuefische/questions/assets/126799897/8b33c2df-e6c4-478d-b653-2cdd501fa19c)

I tried fixing it with various if statements, conditional rendering or ternary operators, but nothing worked. I always run into the same problem: the array is still 0 and can't be read.

- [page with all plants](https://github.com/HenrikeStahlhut/capstone-project/blob/feat/all-plants-overview/pages/plants/all/index.js) 
- [page with plants in each room](https://github.com/HenrikeStahlhut/capstone-project/blob/feat/all-plants-overview/pages/rooms/%5Bid%5D.js)
- Heres my repo: [GitHub](https://github.com/HenrikeStahlhut/capstone-project/tree/feat/all-plants-overview)
- in the localhost, you have to access the page by **/plants/all**

Help would be greatly appreciated 😭

**Comment by mbosselmann** [comment, marked as answer]
Henny and I solved it directly in the conversation and worked with conditional rendering: `{ plant.rooms.length > 0 ? plant.rooms[0].title : "no room is currently assigned” }`
